internals-js/src/operations.ts

```diff
@@ -708,36 +787,31 @@ export class Operation {
   }
 }
 
-function addDirectiveNodesToElement(directiveNodes: readonly DirectiveNode[] | undefined, element: DirectiveTargetElement<any>) {
-  if (!directiveNodes) {
-    return;
-  }
-  const schema = element.schema();
-  for (const node of directiveNodes) {
-    const directiveDef = schema.directive(node.name.value);
-    validate(directiveDef, () => `Unknown directive "@${node.name.value}" in selection`)
-    element.applyDirective(directiveDef, argumentsFromAST(directiveDef.coordinate, node.arguments, directiveDef));
-  }
-}
-
-export function selectionSetOf(parentType: CompositeType, selection: Selection): SelectionSet {
-  const selectionSet = new SelectionSet(parentType);
-  selectionSet.add(selection);
-  return selectionSet;
-}
-
 export class NamedFragmentDefinition extends DirectiveTargetElement<NamedFragmentDefinition> {
+  private _selectionSet: SelectionSet | undefined;
+
   constructor(
     schema: Schema,
     readonly name: string,
     readonly typeCondition: CompositeType,
-    readonly selectionSet: SelectionSet
+    directives?: Directive<any>[],
```

Comment:
Same remark than a bit earlier: the default is incorrect. But yet again, `any` was lazy and I've changed it to `Directive<NamedFragmentDefinition>` (for the record, I'll point that technically we could pass "any" directives to this ctor because even if a directive is attached to a different type of element, the code will copy the directive object. And so `any` is really not incorrect and could provide more flexibility; but that flexibility is not needed right now, and in fairness may never be, so I concede that a more precise typing probably make things more readable for now).